src/naming.js

```diff
@@ -25,12 +25,15 @@ class Naming {
 
   getAlarmName(options) {
     const interpolatedTemplate = options.template
+      .replace('$[stackName]', options.stackName)
       .replace('$[functionName]', options.functionName)
       .replace('$[functionId]', options.functionLogicalId)
       .replace('$[metricName]', options.metricName)
       .replace('$[metricId]', options.metricId);
 
-    return `${options.stackName}-${interpolatedTemplate}`;
+    const prefix = options.stackNamePrefix === false ? '' : `${options.stackName}-`;
```

Comment:
Should we prefix alarm name with stack name when `$[stackName]` is in the template itself? Makes sense for consistency, but pretty weird at the same time. Thoughts?

Reply:
Yeah makes sense to disable the prefix automatically if the var is in the template. I'll update.